src/dynamics/ads_player/ads_player.js

```diff
@@ -381,34 +377,51 @@ Scoped.define("module:Ads.Dynamics.Player", [
                     })
                 },
 
-                setEndCardBackground: function(width, height) {
+                captureAdEndCardBackground: function(width, height) {
                     const ad = this.get("ad");
                     if (ad) {
-                        this.updateEndCardImage(ad, width, height);
+                        this.generateEndCardImage(ad, width, height);
                     }
                 },
 
-                updateEndCardImage: function(ad, width, height) {
+                setEndCardBackground: function(width, height) {
+                    if (width && height) {
+                        if (this.shouldShowFirstFrameAsEndcard()) {
+                            this.captureAdEndCardBackground(width, height);
+                            if (this._src) {
```

Comment:
Sorry just saw, not sure why you need this line? 
As `this._src` could be outdated, and you'll set it on the end as well. 
Will be better to use like this.set(`backgroundsrc`, this._canvas.toDataURL("image/png")) and if needed you can listen to it's changes like this.on(`change:backgroundsrc`, (newSrc) => this.getAdContainer().style.backgroundImage = `url("${newSrc}")`; )
I'm just commenting, if you want can improve.

Reply:
Oop just saw this. It's there because during resizing (assuming the ad completed already) and the video resizes (theoretically unlikely to happen), it would make sense to resize the captured image and set the background with that newly resized image.

The `change:backgroundSrc` does make sense and would accomplish the same thing (in a better way) so will make this change

Reply:
@sambua just made the change and also used `this.parent()._drawFrame(...` for capturing the frame since its an existing method i didn't realize existed